src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -924,6 +924,10 @@ protected void GenerateFindFirstChar()
         {
             _runtextposLocal = DeclareInt32();
             _runtextendLocal = DeclareInt32();
+            if (_code!.RightToLeft)
```

Comment:
It's so easy to read expressions like this as `if (_code != RightToLeft)` 😒

Reply:
I know. 😦 We might be able to avoid it with some well-placed asserts.

Reply:
I think that would be worth doing in cases like this where `!` is sprinkled everywhere. It might actually make it more likely we'd catch a real case because we're not getting inured to `!`'s